cwms-data-api/src/test/java/cwms/cda/api/EntityControllerTestIT.java

```diff
@@ -0,0 +1,294 @@
+
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .accept(Formats.JSONV2)
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .get("/entity/" + ENTITY_ID)
+        .then()
+            .log().ifValidationFails(LogDetail.ALL, true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_BAD_REQUEST));
+    }
+
+    // Entity ID in the URL must match the id.name in the request body
+    @Test
+    void update_mismatched_entity_id_bad_request() throws Exception {
+        TestAccounts.KeyUser user = TestAccounts.KeyUser.SPK_NORMAL;
+        InputStream in = this.getClass().getResourceAsStream("/cwms/cda/data/dto/entity.json");
+        Assertions.assertNotNull(in);
+        String entity = IOUtils.toString(in, java.nio.charset.StandardCharsets.UTF_8);
+
+        // different entity id
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .accept(Formats.JSONV2)
+            .contentType(Formats.JSONV2)
+            .body(entity)
+            .header(AUTH_HEADER, user.toHeaderValue())
+            .queryParam(Controllers.OFFICE, OFFICE_ID)
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .patch("/entity/" + "different")
+        .then()
+            .log().ifValidationFails(LogDetail.ALL, true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_NOT_FOUND));
+
+        // missing office id
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .accept(Formats.JSONV2)
+            .contentType(Formats.JSONV2)
+            .body(entity)
+            .header(AUTH_HEADER, user.toHeaderValue())
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .patch("/entity/" + ENTITY_ID)
+        .then()
+            .log().ifValidationFails(LogDetail.ALL, true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_BAD_REQUEST));
+    }
+
+    // getAll with no query params: must return 200 and a list (empty allowed)
+    @ParameterizedTest
+    @ValueSource(strings = {Formats.JSONV2, Formats.DEFAULT})
+    void getAll_no_params_returns_200_empty_list_ok(String format) {
+        given()
+            .log().ifValidationFails(LogDetail.ALL, true)
+            .accept(format)
+        .when()
+            .redirects().follow(true)
+            .redirects().max(3)
+            .get("/entity")
+        .then()
+            .log().ifValidationFails(LogDetail.ALL, true)
+        .assertThat()
+            .statusCode(is(HttpServletResponse.SC_OK))
+            .body("entities", notNullValue())
+            .body("entities.size()", greaterThanOrEqualTo(0));
+    }
+
+    // Show simple filtering works with getAll and parent entity id only
+    @Test
+    void getAll_with_parent_filter_returns_200_empty_list_ok() {
```

Comment:
Add a test for the match null parents flag. 
1. Create a entity with a null parent
2. GET request with flag set to true -> should get it back
3. GET request with flag set to false -> shouldn't get it back. 
4. delete entity